lib/config-generator.js

```diff
@@ -365,6 +366,25 @@ class ConfigGenerator {
         let splitChunks = {
             chunks: this.webpackConfig.shouldSplitEntryChunks ? 'all' : 'async'
         };
+
+        // hash the split filenames in production to protect exposing entry names
+        if (this.webpackConfig.shouldSplitEntryChunks && this.webpackConfig.isProduction()) {
+            // taken from SplitChunksPlugin
+            const hashFilename = name => {
+                return crypto
+                    .createHash('md4')
+                    .update(name)
+                    .digest('hex')
+                    .slice(0, 8);
+            };
+
+            splitChunks.name = function(module, chunks, cacheGroup) {
+                const names = chunks.map(c => c.name);
+
+                return cacheGroup + '~' + hashFilename(names.join('~'));
```

Comment:
I'm not sure `~` is a good separator choice, it's part of the "unsafe characters" listed in this RFC: http://www.ietf.org/rfc/rfc1738.txt

Reply:
that's the separator used by default in Webpack 4